src/main/java/org/example/hospital/convertors/UserConverter.java

```diff
@@ -0,0 +1,28 @@
+package org.example.hospital.convertors;
+
+import org.example.hospital.dto.UserDTO;
+import org.example.hospital.entity.User;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserConverter implements Converter<User, UserDTO>{
+    @Override
+    public UserDTO convertToDTO(User entity, UserDTO dto) {
+        dto.setUserId(entity.getUserId());
+        dto.setEmail(entity.getEmail());
+        dto.setPassword(entity.getPassword());
+        dto.setRole(entity.getRole());
+        return dto;
+    }
+
+    @Override
+    public User convertToEntity(UserDTO dto, User entity) {
+        if (entity.getUserId() == null) {
```

Comment:
Please remove null checking here too

Reply:
There is no need to set an Id during an update (Id will be overwritten and cause an error), so I have a null checker to set Id only for new entities.

Reply:
You should never set userId on the entity with the convertor. 
If you are creating there is no id on entity and no id on dto
If you are updating there is already id on entity , and no need to set.
You can remove the check and never set the id.